core/app/modules/kit-library/assets/js/components/favorites-actions.js

```diff
@@ -15,19 +15,51 @@ export default function FavoritesActions( props ) {
 					hideText={ true }
 					icon="eicon-heart"
 					className={ `e-kit-library__kit-favorite-actions e-kit-library__kit-favorite-actions--active ${ loadingClasses }` }
-					onClick={ () => ! isLoading && removeFromFavorites.mutate( props.id ) }
+					onClick={ () => {
+						// eslint-disable-next-line no-unused-expressions
+						! isLoading && removeFromFavorites.mutate( props.id );
+						eventTrackingDispatch(
+							'kit-library/mark-as-favorite',
+							{
+								kit_name: props.name,
+								event: 'favorite icon interaction',
+								source: '/' === props.source ? 'home page' : 'overview',
+								event_type: 'search',
+								action: 'uncheck',
+							},
+						)
+					} }
 			/>
 			: <Button
 					text={ __( 'Add to Favorites', 'elementor' ) }
 					hideText={ true }
 					icon="eicon-heart-o"
 					className={ `e-kit-library__kit-favorite-actions ${ loadingClasses }` }
-					onClick={ () => ! isLoading && addToFavorites.mutate( props.id ) }
+					onClick={ () => {
+						// eslint-disable-next-line no-unused-expressions
+						! isLoading && addToFavorites.mutate( props.id );
+						eventTrackingDispatch(
+							'kit-library/mark-as-favorite',
```

Comment:
Same event on both options but different meta?

Reply:
Also, validate the props (like index) before sending. Maybe they do not exist?

Reply:
Yes, same event.
didn't validate bu changed to optional param with default `null` value